src/andromede/model/constraint.py

```diff
@@ -84,3 +86,24 @@ def __init__(
                 self.upper_bound = upper_bound
             else:
                 self.upper_bound = literal(float("inf"))
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Constraint):
+            return False
+        return (
+            self.name == other.name
+            and expressions_equal(self.expression, other.expression)
+            and _expressions_equal_if_present(self.lower_bound, other.lower_bound)
+            and _expressions_equal_if_present(self.upper_bound, other.upper_bound)
+        )
+
+
+def _expressions_equal_if_present(
+    lhs: Optional[ExpressionNode], rhs: Optional[ExpressionNode]
+) -> bool:
+    if lhs is None and rhs is None:
+        return True
+    elif lhs is None or rhs is None:
+        return False
+    else:
+        return expressions_equal(lhs, rhs)
```

Comment:
Do we want to have constraint equality if `lhs is None` and `rhs == +infinity` (for upper bound) for example ?